### DISCUSSION TITLE
`where` in nested `upsert`

### DISCUSSION THREAD

**Discussion by cawa-93** [prisma/prisma#13528] [Q&A]
In my program there are links and Templates for 1-n.
The peculiarity of templates is that they can be private or public.

What I need:
Create a new link, connect it to an existing template, _or_ create a new template.
_OR_
Update an existing link, connect it to an existing template, _or_ create a new template, _or_ update an already connected template but **only if it's private**.

Here is my query that does almost what I want:
```ts
db.shortLink.upsert({
  where: {
    id: input.id || 0,
  },
  create: {
    destinationUrl: input.destinationUrl,
    owner: {
      connect: {id: ctx.user.id},
    },
    template:
      typeof input.template === 'number'
        ? {connect: {id: input.template}}
        : {
          create: {
            private: input.template.isPrivate,
            owner: {
              connect: {id: ctx.user.id},
            },
          },
        },
  },
  update: {
    destinationUrl: input.destinationUrl,
    template:
      typeof input.template === 'number'
        ? {connect: {id: input.template}}
        : {
          upsert: {
            create: {
              private: input.template.isPrivate,
              owner: {
                connect: {id: ctx.user.id},
              },
            },
            update: {
              private: input.template.isPrivate,
              owner: {
                connect: {id: ctx.user.id},
              },
            },
          },
        },
  },
})
```
In general, it works almost as expected. With one exception: It is not possible to prevent the update of _public templates_. I would like to specify an additional parameter `where` in the `shortLink.update.template.upsert`to say
If the connected template meets the condition - update it. Otherwise - create a new one.
```ts
db.shortLink.upsert({
  where: {
    id: input.id || 0,
  },
  create:  { /* ... */ },
  update: {
    destinationUrl: input.destinationUrl,
    template:
      typeof input.template === 'number'
        ? {connect: {id: input.template}}
        : {
          upsert: {
            where: { private: true }, // <-- UPDATE template connected to shortLink only if it's private otherwise create and connect new
            create: { /* ... */ },
            update:  { /* ... */ },
          },
        },
  },
})
```

**Comment by pantharshit00** [comment]
Hello @cawa-93, our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

Can you please share your schema here so that I can try this out in my machine?

**Reply by vladi-stevanovic** [reply]
:wave: Hello @cawa-93 , I’m going through pending GitHub Discussions to check if you need any help. Is this issue still open?

Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! :raised_hands:

**Reply by cawa-93** [reply]
Hi @vladi-stevanovic . Thanks for answer. I actually dont solve problem. But a part of bussines logic was changed and this question no more actual. I think there was something in one of the prisma updates that could solve this, but I haven't tested. Thanks again for your attention

**Reply by vladi-stevanovic** [reply]
Thank you for following up @cawa-93 ! I'm glad to hear you ended up solving the problem otherwise. 
If you come across any other questions or issues don't hesitate to open a new discussion!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
